### ISSUE TITLE
Failed to create record: 401 Unable to authenticate you

### ISSUE THREAD

**Issue by dpatriarche** [digitalocean/terraform-provider-digitalocean#615]
Starting some time in the last couple of months the digitalocean terraform library fails for me when creating a DNS record, with a "401 Unable to authenticate you". The same terraform scripts worked as recently as January 2021. My API key still works for creating Droplets via terraform, and I also verified that the same API key works for creating DNS records via curl.

### Terraform Version
Terraform v0.14.10

### Affected Resource(s)
- digitalocean_record

### Terraform Configuration Files
File: providers.tf
```hcl
terraform {
  required_providers {
    digitalocean = {
      source = "digitalocean/digitalocean"
      version = ">= 2.7.0"
    }
  }
}
```

File: variables.tf
```hcl
variable "digitalocean_api_token" {
  description = "The API key used for the DigitalOcean API"
  type = string
  default = "MY_API_KEY"  # (MY_API_KEY replaces my *real* key)
}
```

File: main.tf
```hcl
resource "digitalocean_record" "hello" {
  domain = "aianalysis.ca"  # A domain controlled by my DO account
  type = "A"
  name = "hello"
  value = "10.0.0.1"
}
```

### Debug Output
https://gist.github.com/dpatriarche/752bd5b70dc9a743a92126a8c8faee63

### Panic Output
n/a

### Expected Behavior
The DNS record should have been created.

### Actual Behavior
The DNS record was not created. he following error was output:

```
Error: Failed to create record: POST https://api.digitalocean.com/v2/domains/aianalysis.ca/records: 401 Unable to authenticate you

  on main.tf line 1, in resource "digitalocean_record" "hello":
   1: resource "digitalocean_record" "hello" {
```

Using curl I can create this record though:

```
$ curl -X POST -H "Content-Type: application/json" -H "Authorization: Bearer $DO_TOKEN" -d '{"type":"A","name":"hello","data":"10.0.0.1","priority":null,"port":null,"ttl":300,"weight":null,"flags":null,"tag":null}' "https://api.digitalocean.com/v2/domains/aianalysis.ca/records"
{"domain_record":{"id":145365489,"type":"A","name":"hello","data":"10.0.0.1","priority":null,"port":null,"ttl":300,"weight":null,"flags":null,"tag":null}}%
```

So the problem seems to be in the digitalocea terraform library.

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
n/a

### References
n/a

**Comment by dpatriarche** [comment]
Just in case his was some problem with my API token, I tried creating a new API token (with write permission). It had the same problem.

**Comment by andrewsomething** [comment]
Hi @dpatriarche, 

Does your Terraform config have a `provider` block? If you are storing the token in the `digitalocean_api_token` variable, I'd expect there to be something like:

```
provider "digitalocean" {
  token = var.digitalocean_api_token
}
```

If the token is not set in the `provider` block, then Terraform will fallback to looking for it in the `DIGITALOCEAN_TOKEN` and `DIGITALOCEAN_ACCESS_TOKEN` environment variables. Is it possible those were configured in previous usage?

**Comment by dpatriarche** [comment]
Arg, yeah that was the problem. The clips I provided in the issue description were actually a cut-down from our actual terraform recipe that includes a main recipe that creates the DNS records plus a module that creates the DO droplet. The module includes the `provider "digitalocean" {}` block but the main recipe does not. Copying the provider block to the main recipe fixes the problem. The weird thing is that _I believe_ that this recipe used to work with the provider block only in the module. Which is syntactically incorrect, but maybe terraform used to allow the main recipe to "inherit" the provider block from the module? Or maybe we happened to have those DIGITALOCEAN_* env vars set.

In any case, the problem was on my end, and I'll close this issue. Thanks for your help @andrewsomething
